packages/flutter/lib/src/widgets/gesture_detector.dart

```diff
@@ -580,6 +580,12 @@ class GestureDetector extends StatelessWidget {
   }
 }
 
+/// Keeps track of the last [PointerDeviceKind] that initiated a gesture.
```

Comment:
I think it's a slightly 'misleading' usage of ValueNotifier since there are no usages of the Listenable interface or listeners etc. Do you intend to expand its usage in the future?

Also, since we were saying we're going this route to avoid having to design a whole generic API around tracking the kind type (since we think it's only useful in the text selection domain), these knowledge shouldn't leak outside of the text selection classes. 

I would just create a listener that composes a Listener above the TextSelectionGestureDetector that doesn't participate in the arena or gestures etc and that only keeps track of the last primary pointer's kind and reports back to the text fields that injected the tracker to the TextSelectionGestureDetector.

Reply:
Ahh, I misunderstood your previous comment :P
I switched now to using `Listener` and it's much cleaner indeed. I'll push it right away.